src/se/vidstige/jadb/managers/PackageManager.java

```diff
@@ -55,6 +55,17 @@ public void install(File apkFile) throws IOException, JadbException {
         verifyOperation("install", apkFile.getName(), result);
     }
 
+    public void forceInstall(File apkFile) throws IOException, JadbException {
+        RemoteFile remote = new RemoteFile("/sdcard/tmp/" + apkFile.getName());
+        device.push(apkFile, remote);
+        InputStream s = device.executeShell("pm", "install","-r", Bash.quote(remote.getPath()));
+        String result = Stream.readAll(s, Charset.forName("UTF-8"));
+        // TODO: Remove remote file
+        s=device.executeShell("rm", "-f",Bash.quote(remote.getPath()));
+        Stream.readAll(s, Charset.forName("UTF-8"));
+        verifyOperation("install", apkFile.getName(), result);
+    }
+
```

Comment:
Hi @fashioncj , I see you used a non-connected email in your commits. Make sure to add the email `me@chenjia.me` to your github account, this will make you show up as [contributor to the jadb project](https://github.com/vidstige/jadb/graphs/contributors).